gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/visitor/ExecutionVisitor.java

```diff
@@ -141,42 +172,54 @@ public EntityFetcherResponse visit(DataFetcherNode dataFetcherNode) {
 
     EntitiesRequest request = requestBuilder.build();
     IEntityFetcher entityFetcher = queryHandlerRegistry.getEntityFetcher(source);
-    return entityFetcher.getEntities(context, request);
+
+    EntitiesRequest totalEntitiesRequest =
+        EntitiesRequest.newBuilder(executionContext.getEntitiesRequest())
+            .clearSelection()
```

Comment:
Optimization: If the limit is not passed down to the EntityFetcher, can we avoid this extra query?